src/vulkan/engine_vulkan.cc

```diff
@@ -93,7 +274,7 @@ Result EngineVulkan::InitializeWithConfig(EngineConfig* config,
     return Result("Vulkan::InitializeWithConfig device handle is null.");
 
   device_ = MakeUnique<Device>(vk_config->device);
-  return InitDeviceAndCreateCommand();
+  return InitDeviceAndCreateCommand({});
```

Comment:
We should be passing the features in here, no? That way we can verify they exist on the provided device? If you need a PhysicalDevice, you can add that to the struct in amber_vulkan.h to be provided as well.

Reply:
Yes, we must pass features here. I only changed this line and let me handle `EngineVulkan::InitializeWithConfig()` in other commits.